src/agent/snippet.h

```diff
@@ -41,7 +41,9 @@
 #include "common/common.h"
 #include "agent/payload.h"
 #include "Buffer.h"
-#include "common/h/arch.h"
+//#include "arch.h"
+
+typedef unsigned int Register;
```

Comment:
We may want to make a note that this type isn't intended to be used outside of the Dyninst internals (see common/src/Types.h) and that the SPI code should be refactored around this.

Reply:
I will include comments for this type. But I think the best way to solve this is to expose codegenAPI from dyninst. As you can see in this snippet.h, this Register type is used by emitter functions. If SPI can use emitter directly from dyninst, this problem of using Dyninst internal can be fixed elegantly.

Reply:
>expose codegenAPI from dyninst

Agreed, but that's a long-term solution. These comments here are good for now. I just wanted to make sure we change this going forward so we don't have type duplication/drift between SPI and Dyninst.